wlauto/instrumentation/perf/__init__.py

```diff
@@ -91,11 +91,8 @@ class PerfInstrument(Instrument):
     ]
 
     def on_run_init(self, context):
-        if not self.device.is_installed('perf') or self.force_install:
-            binary = context.resolver.get(Executable(self, self.device.abi, 'perf'))
-            self.binary = self.device.install(binary)
-        else:
-            self.binary = 'perf'
+        binary = context.resolver.get(Executable(self, self.device.abi, 'perf'))
+        self.binary = self.device.install_if_needed(binary)
```

Comment:
I don't think this logic is equivalent to what has been removed -- if `self.force_install` is set, then the binary should be re-installed regardless of whether it is needed.